src/actix.rs

```diff
@@ -0,0 +1,118 @@
+//! Actix response types for [`ProblemDetails`].
+//!
+//! Requires feature `actix`.
+//!
+//! With the `actix` feature enabled, [`ProblemDetails`] implements [`ResponseError`] using
+//! [`JsonProblemDetails`]. You can also return [`JsonProblemDetails`] to be specific.
+//! If you want to return XML, you can use [`XmlProblemDetails`].
+//!
+//! # Example
+//!
+//! ```rust
+//! use actix_web::{App, web, HttpServer};
+//! use http::StatusCode;
+//! use problem_details::ProblemDetails;
+//!
+//! async fn handler() -> Result<&'static str, ProblemDetails> {
+//!     // always return a problem description
+//!     Err(ProblemDetails::from_status_code(StatusCode::IM_A_TEAPOT)
+//!         .with_detail("short and stout"))
+//! }
+//!
+//! fn main() {
+//!     HttpServer::new(|| {
+//!         App::new()
+//!             .route("/", web::get().to(handler))
+//!     // build and run server...
+//! }
+//! ```
+use actix_web::{
+    web::Json,
+    {HttpResponse, ResponseError}
+};
+use std::fmt::Debug;
+use http::StatusCode;
+
+use crate::ProblemDetails;
+
+#[cfg(feature = "json")]
+use crate::JsonProblemDetails;
+
+#[cfg(feature = "xml")]
+use crate::XmlProblemDetails;
+
+#[cfg(feature = "json")]
+impl<Ext> ResponseError for ProblemDetails<Ext>
+where
+    Ext: serde::Serialize + Clone + Send + Debug,
+{
```

Comment:
I think it would be nice to implement `fn status_code(&self)` as well here:

```suggestion
{
    fn status_code(&self) -> actix_web::http::StatusCode {
        // Due to http crate version mismatches we need to translate the status code
        let status_code = self
            .0
            .status
            .unwrap_or(StatusCode::INTERNAL_SERVER_ERROR)
            .as_u16();
        actix_web::http::StatusCode::from_u16(status_code)
            .expect("Status code should be translatable")
    }
```